source/common/http/conn_pool_grid.cc

```diff
@@ -17,11 +17,20 @@ absl::string_view describePool(const ConnectionPool::Instance& pool) {
 ConnectivityGrid::WrapperCallbacks::WrapperCallbacks(ConnectivityGrid& grid,
                                                      Http::ResponseDecoder& decoder,
                                                      PoolIterator pool_it,
-                                                     ConnectionPool::Callbacks& callbacks)
+                                                     ConnectionPool::Callbacks& callbacks,
+                                                     bool has_early_data, bool should_use_alt_svc)
     : grid_(grid), decoder_(decoder), inner_callbacks_(&callbacks),
       next_attempt_timer_(
           grid_.dispatcher_.createTimer([this]() -> void { tryAnotherConnection(); })),
-      current_(pool_it) {}
+      current_(pool_it), has_early_data_(has_early_data), should_use_alt_svc_(should_use_alt_svc) {
+  if (Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.conn_pool_new_stream_with_early_data_and_alt_svc") &&
+      !should_use_alt_svc) {
+    // If alt_svc is explicitly disabled, there must have been a failed request over Http3 and the
+    // failure must be post-handshake.
```

Comment:
This comment accurately describes why should_use_alt_svc might be true but it's just before a line of code which disables http3 for this request. So how about adding a sentence like "So disable HTTP/3 for this request." to the comment?